Tests/Admin/BreadcrumbsBuilderTest.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Admin;
+
+use Sonata\AdminBundle\Tests\Fixtures\Admin\CommentAdmin;
+use Sonata\AdminBundle\Tests\Fixtures\Admin\PostAdmin;
+use Symfony\Component\HttpFoundation\Request;
+
+class BreadcrumbsBuilderTest extends \PHPUnit_Framework_TestCase
```

Comment:
I will think about how to make this really unitary. The Admin will need unit tests too.